application/squeeze2cast/squeeze2cast.c

```diff
@@ -890,6 +912,9 @@ static void *MainThread(void *args)
 				}
 			}
 		}
+
+		// Called to allow expired devices to be reliably timed out
+		mDNSsearchCallback(NULL, NULL, &ignored);
```

Comment:
I don't think this is needed because the mDNS callback regularly calls even with a NULL s (that(s something I've changed not so long agoà

Reply:
It should have but it's slow. Now, I've been thinking about this and maybe what I should do is split the update in two parts and leave the mDNS callback data process unchanged but have the expired players "checked" being a different function that will be called from the either the callback or the main loop. hat part would be protected then. You're right that a more accurate compliance to user's expiration timeout is better, and if I can do that at no risk, this should be done.

What I can do at this point is accept the PR and do my own mods then push things back, or wait for your proposal. Up to you and thanks for taking the time for doing a PR. Not so many people do and I know this is painful to try to wrap your head around other people's code and especially that code of mine is pretty old and not so pretty actually...

Reply:
Doubt if many people will notice if it takes an extra minute and a half or so to remove an expired renderer. (Before it could have been half a day; that's more problematic.) But you're right that perhaps it's cleaner to separate into an independent function. For now I've reverted the change, and removal will depend upon the callback being called by mdnssd_query. 

Happy to help if I can. Your tools are incredibly useful! You've got to be the world's expert on the chromecast and airplay protocols outside of Google and Apple :)

Reply:
Note that leaving the removal in the callback function means that the update of the config file can also remain in a single location (within the callback).